src/device/router_device_channels_worker.erl

```diff
@@ -184,141 +191,124 @@ init(Args) ->
         chain = Blockchain,
         event_mgr = EventMgrRef,
         device_worker = DeviceWorkerPid,
-        device = Device,
-        fcnt = -1
+        device = Device
     }}.
 
 handle_call(_Msg, _From, State) ->
     lager:warning("rcvd unknown call msg: ~p from: ~p", [_Msg, _From]),
     {reply, ok, State}.
 
 handle_cast(handle_join, State) ->
-    {noreply, State#state{fcnt = -1}};
+    {noreply, State};
 handle_cast({handle_device_update, Device}, State) ->
     {noreply, State#state{device = Device}};
 handle_cast(
-    {handle_frame, Device, #data_cache{pub_key = PubKeyBin, packet = Packet, frame = Frame} = Data,
-        {Balance, Nonce} = BN},
-    #state{data_cache = DataCache0, balance_cache = BalanceCache0, fcnt = CurrFCnt} = State
+    {handle_frame, Device,
+        #data_cache{
+            uuid = UUID,
+            pub_key = PubKeyBin,
+            packet = Packet
+        } = Data},
+    #state{data_cache = DataCache0} = State
 ) ->
-    FCnt = router_device:fcnt(Device),
-    CheckFCnt = FCnt =< CurrFCnt andalso router_device_utils:mtype_to_ack(Frame#frame.mtype) == 0,
-    DataCache1 =
-        case CheckFCnt of
-            true ->
-                lager:debug("we received a late packet ~p from ~p: ~p", [
-                    {FCnt, CurrFCnt},
-                    PubKeyBin,
-                    Packet
-                ]),
-                DataCache0;
-            false ->
-                case maps:get(FCnt, DataCache0, undefined) of
-                    undefined ->
-                        _ = erlang:send_after(?DATA_TIMEOUT, self(), {data_timeout, FCnt}),
-                        maps:put(FCnt, #{PubKeyBin => Data}, DataCache0);
-                    CachedData0 ->
-                        CachedData1 = maps:put(PubKeyBin, Data, CachedData0),
-                        case maps:get(PubKeyBin, CachedData0, undefined) of
-                            undefined ->
-                                maps:put(FCnt, CachedData1, DataCache0);
-                            #data_cache{packet = #packet_pb{signal_strength = RSSI}} ->
-                                case Packet#packet_pb.signal_strength > RSSI of
-                                    true ->
-                                        maps:put(FCnt, CachedData1, DataCache0);
-                                    false ->
-                                        DataCache0
-                                end
-                        end
-                end
+    Action =
+        case maps:get(UUID, DataCache0, undefined) of
+            undefined ->
+                new_uuid;
+            #{PubKeyBin := CachedPacket} = CachedData ->
```

Comment:
why are we still doing this here?

Reply:
Let's always take packets here. Don't care about stronger or weaker rssi